### PR TITLE
Ensuring more atomicity 

### PR DESCRIPTION
**Pull request by wisdomseeker2001**
Chnaged to state.update in all VMs!

### CODE REVIEW

**Review by Viccha18 — APPROVED** [review]
LGTM, I guess this is what happen when a team doesn't know about best practices and we kept copying each other ways of coding things. Thanks for the tedious work ! 👍

File: app/src/main/java/com/github/se/eventradar/viewmodel/AuthenticationViewModel.kt [hunk 1]
```diff
@@ -132,7 +133,9 @@ class LoginViewModel @Inject constructor(private val userRepository: IUserReposi
   ) {
     when (val result = userRepository.addUser(userValues, userId)) {
       is Resource.Success -> {
-        state.value = state.value.copy(isSignUpCompleted = true, isSignUpSuccessful = true)
+        state.update { currentState ->
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/viewmodel/AuthenticationViewModel.kt:136]: Is there a reason why you use the `currentState` here and `it` elsewhere?

File: app/src/main/java/com/github/se/eventradar/viewmodel/MessagesViewModel.kt [hunk 2]
```diff
@@ -58,7 +58,8 @@ constructor(
               }
 
           // Update the UI state with the sorted and filtered message list
-          _uiState.value = _uiState.value.copy(messageList = sortedMessageList)
+
```
Comment (jgeaso1266) [inline app/src/main/java/com/github/se/eventradar/viewmodel/MessagesViewModel.kt:61]: nit: space should be deleted

**Review by jgeaso1266 — COMMENTED** [review]
Nice, but the use of `currentState` vs `it` seems random. I suggest they be the same for consistency. Also, this change should also be made in the tests.

**Pull request merged**